packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -405,6 +388,85 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         emit BatchMintLp(msg.sender, _vaultId, _wPowerPerpAmount, _collateralToMint, _collateralToLP);
     }
 
+    /**
+     * @notice FLash mint short position, LP in Uni v3, use LP NFT as collateral and withdraw ETH collateral to repay flashloan
+     * @dev sender can specify the amount of ETH collateral to withdraw in case vault.collateralAmount > ETH to repay for loan
+     * @param _params FlashloanWMintDepositNftParams struct
+     */
+    function flashloanWMintDepositNft(FlashloanWMintDepositNftParams calldata _params) external payable {
+        _flashLoan(
+            weth,
+            _params.collateralToFlashloan,
+            uint8(CALLBACK_SOURCE.FLASHLOAN_W_MINT_DEPOSIT_NFT),
+            abi.encode(_params)
+        );
+    }
+
+    function _flashCallback(
+        address _initiator,
+        address, /*_asset*/
+        uint256 _amount,
+        uint256 _premium,
+        uint8 _callSource,
+        bytes memory _calldata
+    ) internal override {
+        if (CALLBACK_SOURCE(_callSource) == CALLBACK_SOURCE.FLASHLOAN_W_MINT_DEPOSIT_NFT) {
+            FlashloanWMintDepositNftParams memory data = abi.decode(_calldata, (FlashloanWMintDepositNftParams));
+
+            // convert flashloaned WETH to ETH
+            IWETH9(weth).withdraw(_amount);
+
+            uint256 vaultId = IController(controller).mintWPowerPerpAmount{value: data.collateralToDeposit}(
+                data.vaultId,
+                data.wPowerPerpAmount,
+                0
+            );
+
+            // LP data.wPowerPerpAmount & data.collateralToLp in Uni v3
+            uint256 amount0Desired = isWethToken0 ? data.collateralToLp : data.wPowerPerpAmount;
+            uint256 amount1Desired = isWethToken0 ? data.wPowerPerpAmount : data.collateralToLp;
+            uint256 uniTokenId = _lpWPowerPerpPool(
+                address(this),
+                data.collateralToLp,
+                amount0Desired,
+                amount1Desired,
+                data.lpAmount0Min,
+                data.lpAmount1Min,
+                block.timestamp,
+                data.lpLowerTick,
+                data.lpUpperTick
+            );
+
+            _checkLpMintExcess(vaultId);
+
+            // deposit Uni NFT token in vault
+            INonfungiblePositionManager(nonfungiblePositionManager).approve(controller, uniTokenId);
+            IController(controller).depositUniPositionToken(vaultId, uniTokenId);
+
```

Comment:
IController(controller).withdraw(vaultId, _amount.add(_premium).add(data.collateralToWithdraw));

Reply:
what do u mean ?

Reply:
Here if it is a new vault, that is just well collateralized, user need to send the fee. if it is a vault overcollateralized and fee can be paid from vault, `data.collateralToWithdraw` should have the fee + any other amount user want to withdraw (in case there is any)